test/transpile.jl

```diff
@@ -1897,3 +1897,103 @@ end
 
     @test isequal(transpile(transpiler, valid_circuit), valid_circuit)
 end
+
+@testset "RejectGatesOnExcludedConnectionsTranspiler for all-to-all" begin
+
+    connectivity = AllToAllConnectivity()
+
+    circuit = QuantumCircuit(
+        qubit_count = 2,
+        instructions = [control_z(1, 2)],
+        name = "test-name",
+    )
+    transpiler = RejectGatesOnExcludedConnectionsTranspiler(connectivity)
+
+    @test isequal(transpile(transpiler, circuit), circuit)
+end
+
+@testset "RejectGatesOnExcludedConnectionsTranspiler for line connectivity" begin
+
+    excluded_connections = [(3, 4)]
+    connectivity = LineConnectivity(5, Int[], excluded_connections)
+    transpiler = RejectGatesOnExcludedConnectionsTranspiler(connectivity)
+
+    invalid_circuit = QuantumCircuit(
+        qubit_count = 5,
+        instructions = [sigma_x(2), control_z(3, 4)],
+        name = "test-name",
+    )
+    @test_throws(
+        DomainError(
+            transpiler.connectivity,
+            "the Snowflurry.ControlZ gate on qubits [3, 4] cannot be applied since " *
+            "connection (3, 4) is unavailable",
+        ),
+        transpile(transpiler, invalid_circuit)
+    )
+
+    valid_circuit = QuantumCircuit(
+        qubit_count = 5,
+        instructions = [sigma_x(3), control_z(4, 5)],
+        name = "test-name",
+    )
+
+    @test isequal(transpile(transpiler, valid_circuit), valid_circuit)
+
+    valid_circuit = QuantumCircuit(
+        qubit_count = 5,
+        instructions = [sigma_x(5), control_z(2, 3)],
+        name = "test-name",
+    )
+
+    @test isequal(transpile(transpiler, valid_circuit), valid_circuit)
+end
+
+@testset "RejectGatesOnExcludedConnectionsTranspiler for lattice connectivity" begin
+
+    excluded_connections = [(5, 1)]
+    connectivity = LatticeConnectivity(3, 4, Int[], excluded_connections)
+    transpiler = RejectGatesOnExcludedConnectionsTranspiler(connectivity)
+
+    invalid_circuit = QuantumCircuit(
+        qubit_count = 12,
+        instructions = [sigma_x(6), control_z(1, 5)],
+        name = "test-name",
+    )
+    @test_throws(
+        DomainError(
+            transpiler.connectivity,
+            "the Snowflurry.ControlZ gate on qubits [1, 5] cannot be applied since " *
+            "connection (1, 5) is unavailable",
+        ),
+        transpile(transpiler, invalid_circuit)
+    )
+
+    valid_circuit = QuantumCircuit(
+        qubit_count = 12,
+        instructions = [sigma_x(1), control_z(9, 6)],
+        name = "test-name",
+    )
+    @test isequal(transpile(transpiler, valid_circuit), valid_circuit)
+
+    valid_circuit = QuantumCircuit(
```

Comment:
I dont think we should consider this valid, this implementation is tied to a hardware capability, which cannot process such a gate